File: exportData.py
import os
import urllib.request
import json

env_path = '.env.local'
SUPABASE_URL = ''
SUPABASE_KEY = ''

if os.path.exists(env_path):
    with open(env_path, 'r', encoding='utf-8') as f:
        for line in f:
            if '=' in line:
                key, val = line.strip().split('=', 1)
                if key == 'NEXT_PUBLIC_SUPABASE_URL':
                    SUPABASE_URL = val
                elif key == 'NEXT_PUBLIC_SUPABASE_ANON_KEY':
                    SUPABASE_KEY = val

if not SUPABASE_URL or not SUPABASE_KEY:
    print("Missing Supabase credentials")
    exit(1)

TABLES = [
    'solicitudes_ayuda',
    'personas_desaparecidas',
    'desaparecidos_actualizaciones',
    'rescatados',
    'rescatados_publicos',
    'avisos',
    'traslados'
]

os.makedirs('db_export', exist_ok=True)

for table in TABLES:
    print(f"Exporting table: {table}...")
    url = f"{SUPABASE_URL}/rest/v1/{table}?select=*"
    req = urllib.request.Request(url, headers={
        'apikey': SUPABASE_KEY,
        'Authorization': f'Bearer {SUPABASE_KEY}'
    })
    
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            data = json.loads(response.read().decode())
            if data:
                file_path = f"db_export/{table}.json"
                with open(file_path, 'w', encoding='utf-8') as out:
                    json.dump(data, out, indent=2, ensure_ascii=False)
                print(f"✅ Saved {len(data)} records to {file_path}")
            else:
                print(f"⚠️ No data found for {table}")
    except Exception as e:
        print(f"Failed to fetch {table}: {e}")

print("🎉 Export finished!")

File: scripts/js/exportData.js
const fs = require('fs');
const path = require('path');

// Directorio raiz del proyecto (scripts/js/ -> ../../)
const PROJECT_ROOT = path.resolve(__dirname, '../../');

// Manually parse .env.local
const envPath = path.join(PROJECT_ROOT, '.env.local');
let SUPABASE_URL = process.env.NEXT_PUBLIC_SUPABASE_URL;
let SUPABASE_KEY = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

if (fs.existsSync(envPath)) {
  const envContent = fs.readFileSync(envPath, 'utf8');
  envContent.split('\n').forEach(line => {
    const match = line.match(/^([^=]+)=(.*)$/);
    if (match) {
      const key = match[1].trim();
      const value = match[2].trim();
      if (key === 'NEXT_PUBLIC_SUPABASE_URL') SUPABASE_URL = value;
      if (key === 'NEXT_PUBLIC_SUPABASE_ANON_KEY') SUPABASE_KEY = value;
    }
  });
}

const TABLES = [
  'solicitudes_ayuda',
  'personas_desaparecidas',
  'desaparecidos_actualizaciones',
  'rescatados',
  'rescatados_publicos',
  'avisos',
  'traslados'
];

const { createClient } = require('@supabase/supabase-js');
const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);

async function exportData() {
  const exportDir = path.join(PROJECT_ROOT, 'db_export');
  if (!fs.existsSync(exportDir)){
    fs.mkdirSync(exportDir);
  }

  for (const table of TABLES) {
    console.log(`Exporting table: ${table}...`);
    try {
      let allData = [];
      let from = 0;
      const batchSize = 1000;
      
      while (true) {
        const { data, error } = await supabase
          .from(table)
          .select('*')
          .range(from, from + batchSize - 1);
          
        if (error) {
          console.error(`Error exporting ${table}:`, error.message);
          break;
        }
        
        if (!data || data.length === 0) {
          break;
        }
        
        allData = allData.concat(data);
        if (data.length < batchSize) {
          break;
        }
        from += batchSize;
      }
      
      if (allData.length > 0) {
        const filePath = path.join(exportDir, `${table}.json`);
        fs.writeFileSync(filePath, JSON.stringify(allData, null, 2));
        console.log(`✅ Saved ${allData.length} records to ${filePath}`);
      } else {
        console.log(`⚠️ No data found or accessible for table ${table}.`);
      }
    } catch(err) {
      console.error(`Failed to fetch ${table}:`, err.message);
    }
  }
  
  console.log("🎉 Export finished!");
}

exportData();
